app/controllers/ReturnsController.scala

```diff
@@ -49,183 +49,162 @@ trait ReturnsController extends BaseController with SessionProfile {
+          returnsService.retrieveCTActiveDate flatMap { ctActiveDate =>
+            voluntaryDateForm.bindFromRequest.fold(
+              errors => {
+                val dynamicDate = timeService.dynamicFutureDateExample()
+                Future.successful(BadRequest(VoluntaryStartDatePage(errors, ctActiveDate, dynamicDate)))
+              },
+              success => returnsService.saveVoluntaryStartDate(success._1, success._2, ctActiveDate) map redirectBasedOnReclaim
+            )
+          }
         }
-      }
   }
 
   val mandatoryStartPage: Action[AnyContent] = isAuthenticatedWithProfile {
-    implicit request => implicit profile =>
-      ivPassedCheck {
-        startDateGuard(pageVoluntary = false) {
-          profile.incorporationDate.fold(Future.successful(Ok(MandatoryStartDateConfirmationPage()))) {
-            incorpDate =>
-              returnsService.retrieveMandatoryDates map { dateModel =>
-                val form = MandatoryDateForm.form(incorpDate, dateModel.calculatedDate)
-                val dynamicDate = timeService.dynamicFutureDateExample()
-                Ok(MandatoryStartDateIncorpPage(
-                  dateModel.selected.fold(form) { selection =>
-                    form.fill((selection, dateModel.startDate))
-                  },
-                  dateModel.calculatedDate.format(MonthYearModel.FORMAT_D_MMMM_Y),
-                  dynamicDate
-                ))
-              }
+    implicit request =>
+      implicit profile =>
+        ivPassedCheck {
+          startDateGuard(pageVoluntary = false) {
+            Future.successful(Ok(MandatoryStartDateConfirmationPage()))
           }
         }
-      }
   }
 
   val submitMandatoryStart: Action[AnyContent] = isAuthenticatedWithProfile {
-    implicit request => implicit profile =>
-      ivPassedCheck {
-        profile.incorporationDate match {
-          case Some(incorpDate) =>
-            returnsService.retrieveCalculatedStartDate flatMap { calcDate =>
-                MandatoryDateForm.form(incorpDate, calcDate).bindFromRequest.fold(
-                  errors => {
-                    val dynamicDate = timeService.dynamicFutureDateExample()
-                    Future.successful(BadRequest(MandatoryStartDateIncorpPage(errors, calcDate.format(MonthYearModel.FORMAT_D_MMMM_Y), dynamicDate)))
-                  },
-                  success => returnsService.saveVatStartDate(if (success._1 == DateSelection.calculated_date) Some(calcDate) else success._2) map redirectBasedOnReclaim
-                )
-            }
-          case None => returnsService.saveVatStartDate(None) map redirectBasedOnReclaim
+    implicit request =>
+      implicit profile =>
+        ivPassedCheck {
+          returnsService.saveVatStartDate(None) map redirectBasedOnReclaim
         }
-      }
   }
 
   private def redirectBasedOnReclaim(returns: Returns): Result = {
```

Comment:
Givn the submitMandatoryStart method is much smaller now, this logic should be moved into that method to improve readability